### PR TITLE
fix broken metatensor documentation links

### PR DESCRIPTION
**Pull request by PicoCentauri**
----
📚 Documentation preview 📚: https://rascaline--275.org.readthedocs.build/en/275/

### CODE REVIEW

**Comment by Luthaf** [comment]
I think I would rather have `https://lab-cosmo.github.io/metatensor/` redirect to `https://lab-cosmo.github.io/metatensor/latest/` on metatensor side. This should be doable with a minimal index.html which directly redirects (I though we already did something like this)

**Comment by PicoCentauri** [comment]
A redirect is also fine for me.

File: docs/src/get-started/installation.rst [hunk 1]
```diff
@@ -175,5 +175,5 @@ Other useful configuration options are:
 |                                        | install metatensor-torch?                     |                |
 +----------------------------------------+-----------------------------------------------+----------------+
 
-.. _C++ interface of metatensor: https://lab-cosmo.github.io/metatensor/latest/get-started/installation.html#installing-the-c-and-c-library
-.. _TorchScript interface of metatensor: https://lab-cosmo.github.io/metatensor/latest/get-started/installation.html#for-usage-from-c
+.. _C++ interface of metatensor: https://lab-cosmo.github.io/metatensor/get-started/installation.html#installing-the-c-and-c-library
+.. _TorchScript interface of metatensor: https://lab-cosmo.github.io/metatensor/get-started/installation.html#for-usage-from-c
```
Comment (Luthaf) [inline docs/src/get-started/installation.rst:179]: I think these still need to contain the `latest`. The redirect only happen on direct access to the `metatensor/` url.

Comment (PicoCentauri) [reply docs/src/get-started/installation.rst]: Yes, right.

**Review by Luthaf — APPROVED** [review]

**Pull request merged**